lib/thrift/protocols/binary.ex

```diff
@@ -23,29 +23,21 @@ defmodule Thrift.Protocols.Binary do
   @set 14
   @list 15
 
-  @types %{bool: @bool,
-           byte: @byte,
-           double: @double,
-           i8: @byte,
-           i16: @i16,
-           i32: @i32,
-           i64: @i64,
-           string: @string,
-           binary: @string,
-           struct: @struct,
-           map: @map,
-           set: @set,
-           list: @list
-          }
-
-  for {atom_type, int_type} <- @types do
-    def int_type(unquote(atom_type)) do
-      unquote(int_type)
-    end
-  end
-  def int_type({:map, _}), do: 13
-  def int_type({:set, _}), do: 14
-  def int_type({:list, _}), do: 15
+  @spec type_id(atom | {atom, atom}) :: type_id
+  defp type_id(:bool),      do: @bool
+  defp type_id(:byte),      do: @byte
+  defp type_id(:i16),       do: @i16
+  defp type_id(:i32),       do: @i32
+  defp type_id(:i64),       do: @i64
+  defp type_id(:double),    do: @double
+  defp type_id(:string),    do: @string
+  defp type_id(:struct),    do: @struct
+  defp type_id(:map),       do: @map
+  defp type_id(:set),       do: @set
+  defp type_id(:list),      do: @list
+  defp type_id({:map, _}),  do: @map
+  defp type_id({:set, _}),  do: @set
+  defp type_id({:list, _}), do: @list
```

Comment:
Or if these are the correct patterns, should the non-tuple versions of the container types be removed instead?